lnrpc/rpc_utils.go

```diff
@@ -17,8 +17,30 @@ func RPCTransactionDetails(txns []*lnwallet.TransactionDetail) *TransactionDetai
 
 	for i, tx := range txns {
 		var destAddresses []string
-		for _, destAddress := range tx.DestAddresses {
-			destAddresses = append(destAddresses, destAddress.EncodeAddress())
+		// Re-package destination output information.
+		var outputDetails []*OutputDetail
+		for _, o := range tx.OutputDetails {
+			// Note: DestAddresses is deprecated but we keep
+			// populating it with addresses for backwards
+			// compatibility.
+			for _, a := range o.Addresses {
+				destAddresses = append(destAddresses,
+					a.EncodeAddress())
+			}
+
+			var address string
+			if len(o.Addresses) == 1 {
+				address = o.Addresses[0].EncodeAddress()
```

Comment:
yeah so following up on my comment from before, what about the other addresses here?

Reply:
The `DestAddresses` field is a bit weird, because if an output is not re-presentable by an address (e.g. due to `NonStandard` type or the `MultiSigType`), then the indices in the `DestAddresses` do not correspond to the outputs.

The reason that `MultiSigType` returns a slice of addresses, is because this is the case where the multisig script is directly put inside the scriptPubKey, hence not behind a hash. Therefore it cannot be represented by an address. It is questionable if a slice of addresses for the pubkeys referred to in the multisig script is helpful though. :)

Reply:
ok cool, chatted to oli about this and he agrees with you 👍